### PR TITLE
BUG: Series.any and Series.all for empty or all-null pyarrow-backed dtypes

### LINKED ISSUE #51624: BUG?: Series.any / Series.all behavior for pyarrow vs. non-pyarrow empty/all-NA
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

for dtype in ["bool", "boolean", "boolean[pyarrow]"]:
    ser = pd.Series([], dtype=dtype)
    print(dtype, ser.any(), ser.all())
```

### Issue Description

```
bool False True
boolean False True
boolean[pyarrow] <NA> <NA>
```

At the moment `Series.any` and `Series.all` have inconsistent behavior for pyarrow vs. non-pyarrow dtypes for empty or all null data. The pyarrow behavior is coming from `pyarrow.compute.any` and `pyarrow.compute.all` which return null in these cases. I suspect different behavior here within pandas is likely to be problematic and confusing. Is it OK to diverge from pyarrow here in order to be more consistent within pandas?

### Expected Behavior

Probably to match the behavior of non-pyarrow types

### Installed Versions

.

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] closes #51624
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -198,7 +198,7 @@ Sparse
 
 ExtensionArray
 ^^^^^^^^^^^^^^
--
+- Bug in :meth:`Series.any` and :meth:`Series.all` for empty or all null pyarrow-backed dtypes (:issue:`51624`)
```
Comment (phofl) [inline doc/source/whatsnew/v2.1.0.rst:201]: Can you explain what the bug was? Had to run an example to understand it. Otherwise lgtm

**Review by phofl — APPROVED** [review]
thx

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Comment by phofl** [comment]
back porting as well, because we need it for the other pr

**Comment by phofl** [comment]
@meeseeksdev backport 2.0.x

**Pull request merged**